src/ds5/ds5-motion.cpp

```diff
@@ -178,75 +183,138 @@ namespace librealsense
+            std::unique_ptr<frame_timestamp_reader>(new global_timestamp_reader(std::move(iio_hid_ts_reader), _tf_keeper, enable_global_time_option)),
+            std::unique_ptr<frame_timestamp_reader>(new global_timestamp_reader(std::move(custom_hid_ts_reader), _tf_keeper, enable_global_time_option)),
+            fps_and_sampling_frequency_per_rs2_stream,
+            sensor_name_and_hid_profiles,
+            this);
+
+        auto smart_hid_ep = std::make_shared<ds5_hid_sensor>("Motion Module", hid_ep, this, this);
 
         hid_ep->register_option(RS2_OPTION_GLOBAL_TIME_ENABLED, enable_global_time_option);
-        hid_ep->register_pixel_format(pf_accel_axes);
-        hid_ep->register_pixel_format(pf_gyro_axes);
+
+        // motion correction handling
+        float3x3 imu_to_depth;
+        try {
+            imu_to_depth = _mm_calib->imu_to_depth_alignment();
+        }
+        catch (const std::exception& ex) {
+            LOG_INFO("Motion Module - no extrinsic calibration, " << ex.what());
+        }
+        auto align_imu_axes = [hid_ep, imu_to_depth, this](frame_interface* fr)
+        {
+            if (fr->get_stream()->get_format() != RS2_FORMAT_MOTION_XYZ32F)
+                return;
+
+            auto xyz = (float3*)(fr->get_frame_data());
+
+            try
+            {
+                auto accel_intrinsic = *_accel_intrinsic;
+                auto gyro_intrinsic = *_gyro_intrinsic;
+
+                bool enable_motion_correction = false;
+                if (hid_ep->supports_option(RS2_OPTION_ENABLE_MOTION_CORRECTION))
+                {
+                    auto&& motion_correction_opt = hid_ep->get_option(RS2_OPTION_ENABLE_MOTION_CORRECTION);
+                    enable_motion_correction = motion_correction_opt.is_enabled();
+                }
+
+                if (enable_motion_correction)
+                {
+                    
+                    rs2_stream stream = fr->get_stream()->get_stream_type();
+                    if (stream == RS2_STREAM_ACCEL)
+                        *xyz = (accel_intrinsic.sensitivity * (*xyz)) - accel_intrinsic.bias;
+
+                    if (stream == RS2_STREAM_GYRO)
+                        *xyz = gyro_intrinsic.sensitivity * (*xyz) - gyro_intrinsic.bias;
+                }
+            }
+            catch (const std::exception& ex)
+            {
+                LOG_INFO("Motion Module - no intrinsic calibration, " << ex.what());
+            }
+
+            // The IMU sensor orientation shall be aligned with depth sensor's coordinate system
+            *xyz = imu_to_depth * (*xyz);
+            
+        };
+        frame_callback_ptr post_process_cb = std::make_shared<internal_frame_callback<decltype(align_imu_axes)>>(align_imu_axes);
+
+        // register pre-processing
+        smart_hid_ep->register_processing_block(
+            { {RS2_FORMAT_MOTION_XYZ32F, RS2_STREAM_ACCEL} },
```

Comment:
Is the motion_correction modules implemented here?

Reply:
Please see the lambda in line 210 in the same file.